projects/workforce-allocations-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/AllocationDemandService.kt

```diff
@@ -134,6 +136,23 @@ class AllocationDemandService(
         return AllocationImpact(person.crn, person.name(), staff.toStaffMember(ldapService.findEmailForStaff(staff)))
     }
 
+    fun getDetails(requests: AllocationDetailRequests): AllocationDetails {
+        val cases = personRepository.findAllByCrnAndSoftDeletedFalse(requests.cases.map { it.crn })
+            .associateBy { it.crn }
+        val staff = staffRepository.findAllByCodeIn(requests.cases.map { it.staffCode })
+            .map { it.toStaffMember() }
+            .associateBy { it.code }
+        return AllocationDetails(
+            requests.cases.map {
```

Comment:
Could you use `mapNotNull` here, to save making the crn and name nullable?

Reply:
Makes sense, so they might send in a CRN/Staff pair but one or the other might not exist.